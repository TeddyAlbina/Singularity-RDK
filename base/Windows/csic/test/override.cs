class A {
	public virtual string Identity() { return "A.Identity(): " + ToString(); }
	public void print() { System.Console.WriteLine(Identity()); }
}
class B: A {
	override public string ToString() { return "B.ToString()"; }
	override public string Identity() { return "B.Identity(): " + base.ToString(); }
}
class test {
	public static void Main() {
		A a = new A();
		a.print();
		B b = new B();
		b.print();
		System.Console.WriteLine("{0}", ((A)b).Identity());
		System.Console.WriteLine("{0}", b.Identity());
		((A)b).print();
	}
}
