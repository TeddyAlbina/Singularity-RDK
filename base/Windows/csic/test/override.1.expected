A.Identity(): A
B.Identity(): B
B.Identity(): B
B.Identity(): B
B.Identity(): B
